### PR TITLE
Feature: Add quorum to lvg change_vg

### PR DESCRIPTION
**Pull request by AIXperts**
Added 'quorum' option to lvg.

Signed-off-by: Henrik Morsing <henrik.morsing@aixperts.co.uk>

### CODE REVIEW

**Comment by AIXperts** [comment]
I have tried tidying up now.

File: plugins/modules/lvg.py [hunk 1]
```diff
@@ -292,6 +297,9 @@ def make_vg(module, vg_name):
     pvs = module.params['pvs']
     opt = build_vg_opts(module)
 
+    if quorum:
+        result['msg'] += "Attribute quorom is not supported while changing volume group %s.\n" % vg_name
+
```
Comment (schamola) [inline plugins/modules/lvg.py]: Please move this conditional check to the main function to facilitate fast failure and use module.fail_json to return the error message and exit. You should include this check at line no 791 in the main function of original file. Also, you need to initialise the quorum variable before using it.

File: plugins/modules/lvg.py [hunk 2]
```diff
@@ -179,6 +179,11 @@
     - Specifies the major number of the volume group that is created.
     - Can be used only when creating a volume group, hence when I(state=present).
     type: int
+  quorum:
+    description:
+    - Enables/disables quorum on the volume group.
+    - Can be used changing a volume group, hence when I(state=present).
```
Comment (schamola) [inline plugins/modules/lvg.py]: Use this statement instead: "Can be used while changing an existing volume group."

**Review by schamola — CHANGES_REQUESTED** [review]

**Comment by schamola** [comment]
@AIXperts Please send a test output for your changes.

**Comment by AIXperts** [comment]
Good morning,

Test results below:

[ ~]$ ansible server -i /tmp/hosts -m ibm.power_aix.vg -a "state=present vg_name=rootvg quorum=true"
[WARNING]: Platform aix on host server is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python
interpreter could change this. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
server | CHANGED => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": true,
    "cmd": "chvg -Q y rootvg ",
    "msg": "Volume group rootvg modified.\n",
    "rc": 0,
    "stderr": "",
    "stderr_lines": [],
    "stdout": "Physical Volume   Mirror Pool       \nhdisk0            None              \n",
    "stdout_lines": [
        "Physical Volume   Mirror Pool       ",
        "hdisk0            None              "
    ]
}
[ ~]$ ansible server -i /tmp/hosts -m ibm.power_aix.vg -a "state=present vg_name=rootvg quorum=true"
[WARNING]: Platform aix on host server is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python
interpreter could change this. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
server | SUCCESS => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "cmd": "lsvg -P rootvg",
    "msg": "No changes were needed on volume group rootvg.",
    "rc": 0,
    "stderr": "",
    "stderr_lines": [],
    "stdout": "Physical Volume   Mirror Pool       \nhdisk0            None              \n",
    "stdout_lines": [
        "Physical Volume   Mirror Pool       ",
        "hdisk0            None              "
    ]
}
[ ~]$ ansible da1017 -i /tmp/hosts -m ibm.power_aix.vg -a "state=present vg_name=rootvg quorum=false"
[WARNING]: Platform aix on host server is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python
interpreter could change this. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
server | CHANGED => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": true,
    "cmd": "chvg -Q n rootvg ",
    "msg": "Volume group rootvg modified.\n",
    "rc": 0,
    "stderr": "",
    "stderr_lines": [],
    "stdout": "Physical Volume   Mirror Pool       \nhdisk0            None              \n",
    "stdout_lines": [
        "Physical Volume   Mirror Pool       ",
        "hdisk0            None              "
    ]
}
[ ~]$ ansible server -i /tmp/hosts -m ibm.power_aix.vg -a "state=present vg_name=rootvg quorum=false"
[WARNING]: Platform aix on host server is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python
interpreter could change this. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
server | SUCCESS => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "cmd": "lsvg -P rootvg",
    "msg": "No changes were needed on volume group rootvg.",
    "rc": 0,
    "stderr": "",
    "stderr_lines": [],
    "stdout": "Physical Volume   Mirror Pool       \nhdisk0            None              \n",
    "stdout_lines": [
        "Physical Volume   Mirror Pool       ",
        "hdisk0            None              "
    ]
}

**Comment by AIXperts** [comment]
Hi Schamola,

I don't understand what I am requested to change? And what the above output means, it does not appear top be related to my changes.

Regards,
Henrik

**Comment by nitismis** [comment]
Hi @AIXperts 
I believe you have already made the changes that are requested, so we are fine there.
The output you are seeing is the travis build output as it is running some tests.
I see it is failing 9 testcases which you can find in tests directory as test_lvg.py
We are right now working to find the reason behind failed testcases. We will update here, if or not it is because of your changes.

**Comment by nitismis** [comment]
Hi @AIXperts 
You need to do a small change in **tests/unit/plugins/modules/test_lvg.py**
The change is:
--- test_new.py	2022-10-28 18:04:43.000000000 +0530
+++ test_lvg.py	2022-10-28 17:49:52.000000000 +0530
@@ -34,7 +34,8 @@
     "multi_node_vary": None,
     "auto_on": None,
     "retry": None,
\-    "major_num": None
\+    "major_num": None,
\+    "quorum": None
 }

You have to add **"quorum": None** in the params dictionary and remember to put a **"comma (,)"** in the end if major_num line. As you have included a new attribute "quorum" it is failing some of the unit test cases. It's a small change and this will fix the failing issue.

Thanks and Regards,
Nitish K Mishra

**Review by nitismis — APPROVED** [review]
Now looks good to me. All test cases are passing

File: plugins/modules/lvg.py (hunk 1 above)
Comment (nitismis) [reply plugins/modules/lvg.py]: This is resolved.

File: plugins/modules/lvg.py (hunk 2 above)
Comment (nitismis) [reply plugins/modules/lvg.py]: This is resolved.

**Review by schamola — APPROVED** [review]

**Pull request merged**